hibernate-core/src/main/java/org/hibernate/loader/Loader.java

```diff
@@ -1896,6 +1897,25 @@ private void loadFromResultSet(
 				session.getPersistenceContextInternal().addEntity( euk, object );
 			}
 		}
```

Comment:
There is a block above what is displayed here, lines 1877-1899, that deals with the `AssociationType`s returned by `#getOwnerAssociationTypes()`.

It appears that in the case where an entity association is join fetched, and its `AssociationType#getRHSUniqueKeyPropertyName != null`, an entry will be added in both the above and below block.

I think that what has been added below was intended to be accomplished by the block above. The problem seems to be that `ownerAssociationTypes[i]` will be `null` unless the association is join fetched.

Reply:
FWIW, there were no test failures commenting out the block above, and it did not change the SQL that got generated by hibernate-core tests.

@Steve, do you see a reason to keep the block above?